python/tank/bootstrap/manager.py

```diff
@@ -516,7 +516,21 @@ def prepare_engine(self, engine_name, entity):
 
         return path
 
-    def get_pipeline_configurations(self, project):
+    def get_resolved_pipeline_configuration_descriptor(self, project_id=None):
+        """
+        Resolves a pipeline configuration and returns its associated descriptor object.
```

Comment:
Ah... yes. I think I got it now. Is that because we also have to deal with the fallback, which doesn't have an id and could be any value?